### PR TITLE
BUG: Detect non-positive-definite covariance in GaussianMembershipFunction

### LINKED ISSUE #3589: `itk::itkGaussianMembershipFunction` not triggering exception on negative determinant values
### Description

The condition

```
double det = inv_cov.determinant_magnitude();

if (det < 0.)
{
  itkExceptionMacro(<< "det( m_Covariance ) < 0");
}
```
in [itkGaussianMembershipFunction.hxx:116](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Numerics/Statistics/include/itkGaussianMembershipFunction.hxx#L116) apparently tries to tell whether the determinant of the covariance is negative. However, when a test was added to verify this condition in PR #3581, the test failed. Having a closer look, if the `VNL` method `determinant_magnitude` computes the magnitude of the determinant, and not the determinant, the value will always be positive, and the condition will never be true. However, the documentation of the `VNL` method says that it computes the determinant; also if that is true, if the covariance matrix is a negative scalar (e.g. -2), the determinant should be negative, and it is always non-negative.

Thus:
- The documentation of the VNL method is not accurate.
- The itkGaussianMembershipFunction.hxx:116 should compute the determinant, and not its magnitude (if the computations below do require a non-negative value, at least)

### Steps to Reproduce

1. Get ITK
3. Configure and build ITK
4. 3. Add a test to the itkGaussianMembershipFunctionTest file, after line 52```  // Test negative covariance matrix determinant exception
  covariance.SetSize(MeasurementVectorSize, MeasurementVectorSize);
  covariance.Fill(-1);
  ITK_TRY_EXPECT_EXCEPTION(function->SetCovariance(covariance));```4. Execute the test and check that it does not trigger the expected exception.

### Expected behavior

The condition is useful and an exception is triggered for negative values, or else, the condition is not necessary.

### Actual behavior

The expected exception for covariance matrices with negative determinant values is not triggered because it is its magnitude that it is being tested.

### Reproducibility

100%.

### Versions

`master`.

### Environment

All platforms failed when the test was added.

### Additional Information

Did not study what the determinant_magnitude computes in reality.

### PR DESCRIPTION
**Pull request by hjmjohnson**
## Summary

Fixes #3589.

`GaussianMembershipFunction::SetCovariance()` previously checked

```cpp
const vnl_matrix_inverse<double> inv_cov(m_Covariance.GetVnlMatrix());
const double det = inv_cov.determinant_magnitude();
if (det < 0.) { itkExceptionStringMacro(\"det( m_Covariance ) < 0\"); }
```

`vnl_matrix_inverse::determinant_magnitude()` returns the product of singular values, which is **always non-negative**, so the `det < 0.` check is unreachable dead code. A singular or negative-definite covariance matrix would silently slip through and propagate NaNs into the multivariate Gaussian normalization constant downstream (`m_PreFactor = 1.0 / (std::sqrt(det) * ...)`).

This PR replaces the check with `vnl_determinant()` on the *original* (non-inverted) covariance and rejects `det <= 0`, which catches both singular (`det == 0`) and non-positive-definite (`det < 0`) cases. The cost is `O(n^3)` but `n` is the measurement-vector dimension (typically 1..16), negligible compared to the SVD already computed for the inverse on the next line.

## Diff

```diff
+#include \"vnl/algo/vnl_determinant.h\"
 ...
-  // the determinant is then costless this way
-  const double det = inv_cov.determinant_magnitude();
-
-  if (det < 0.)
-  {
-    itkExceptionStringMacro(\"det( m_Covariance ) < 0\");
-  }
+  // Compute the *signed* determinant of the covariance matrix.  See
+  // commit message for the rationale (determinant_magnitude is the
+  // product of singular values and is never negative).
+  const double det = vnl_determinant(m_Covariance.GetVnlMatrix());
+
+  if (det <= 0.0)
+  {
+    itkExceptionMacro(\"Covariance matrix must be positive definite (det = \" << det << \").\");
+  }
```

## Tests

Three new regression cases added to `itkGaussianMembershipFunctionTest.cxx`:

1. **Singular covariance** `[[1,1],[1,1]]` (det = 0) — must throw.
2. **Negative-definite (indefinite) covariance** `[[1,2],[2,1]]` (det = −3) — must throw.
3. **Well-formed covariance** `[[2,0.5],[0.5,2]]` (det = 3.75, symmetric positive-definite) — must succeed.

## AI Assistance

Generated by [Claude Code](https://claude.com/claude-code) as part of the 2026-04-10 open-issue triage cleanup.

## Test plan

- [x] Local build (`ITKStatisticsTestDriver`) compiles cleanly
- [x] `itkGaussianMembershipFunctionTest` passes locally with the new singular / negative-definite / positive-definite assertions
- [x] `pre-commit run` clean (clang-format, kw-pre-commit)
- [ ] CI: Linux, Windows, macOS, Python, ARM

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@greptileai review this draft before I make it official

File: Modules/Numerics/Statistics/test/itkGaussianMembershipFunctionTest.cxx [hunk 1]
```diff
@@ -89,6 +89,46 @@ itkGaussianMembershipFunctionTest(int, char *[])
...
+  // Singular: rank 1, det = 0
+  NPDMembershipFunctionType::CovarianceMatrixType singular;
+  singular.SetSize(2, 2);
+  singular(0, 0) = 1.0;
+  singular(0, 1) = 1.0;
+  singular(1, 0) = 1.0;
+  singular(1, 1) = 1.0;
+  ITK_TRY_EXPECT_EXCEPTION(npdFunction->SetCovariance(singular));
+
+  // Negative-definite (not symmetric positive-definite): det = -3
```
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply Modules/Numerics/Statistics/test/itkGaussianMembershipFunctionTest.cxx]: Fixed in d11739fdc6 — changed comment to "Indefinite (not symmetric positive-definite)" since eigenvalues of [[1,2],[2,1]] are 3 and -1 (mixed sign = indefinite, not negative-definite).

**Review by thewtex — APPROVED** [review]
